inventory-api/src/main/scala/com/normation/inventory/services/provisionning/CheckInventoryDigest.scala

```diff
@@ -152,7 +152,8 @@ class InventoryDigestServiceV1(
   def getKey (receivedInventory  : InventoryReport) : Box[(PublicKey, KeyStatus)] = {
     for {
       nodeInventory <- repo.get(receivedInventory.node.main.id) match {
-        case Full(storedInventory) =>
+        // Don't get the node inventory if there is no public key in it
+        case Full(storedInventory) if ! storedInventory.node.publicKeys.isEmpty =>
```

Comment:
I don't understand why you need to check here for emptiness, as it is done just after - and in the following check the error message would be more specialised. 
In fact, I think that you want to check something more powerfull than just emptiness, it's the key validity: "if there is an existing stored inventory with at least (exactly ?) one readable key in it, then use it; Else, consider the stored inventory like not intersting, and use the new one key.